tests/cephadm/test_replace_osd.py

```diff
@@ -0,0 +1,85 @@
+from ceph.ceph_admin import CephAdmin
+from ceph.ceph_admin.common import fetch_method
+from ceph.ceph_admin.daemon import Daemon
+from cli.cephadm.cephadm import CephAdm
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def replace_osd(ceph_cluster, osd_id, node_name, base_cmd):
+    installer_node = ceph_cluster.get_nodes(role="installer")[0]
+    cephadm = CephAdm(installer_node)
+    log.info("Replacing OSD {}".format(osd_id))
+    try:
+        cephadm.ceph.orch.osd.rm(osd_id=osd_id, replace=True)
+        log.info("OSD {} has been replaced successfully".format(osd_id))
+    except Exception as e:
+        log.error("Error replacing OSD {}: {}".format(osd_id, e))
+        return 1
+
+
+def add_osd(ceph_cluster, node_name, command, daemon, add_osd_config):
+    log.info("Adding OSD on {}".format(node_name))
+    try:
+        method = fetch_method(daemon, command)
+        method(add_osd_config)
+    except Exception as e:
+        log.error(f"Error adding OSD: {e}")
+        return 1
+
+
+def check_osd_id(osd_id, installer, cephadm):
+    # Get the OSD ID of the new OSD after the replacement
+    cmd = f"ceph osd tree | grep osd.{osd_id} | cut -d' ' -f2"
+    try:
+        cephadm.shell([cmd])
+    except Exception as e:
+        print(f"Failed to execute command {cmd}: {e}")
```

Comment:
Same error message will be raised by `exec` method, no need to catch exception here.